server/controllers/ticket/edit-comment.php

```diff
@@ -60,14 +60,19 @@ public function handler() {
             $ticket = Ticket::getByTicketNumber(Controller::request('ticketNumber'));
         }
 
-        $ticketEventArray = $ticketEvent->toArray();
+        if (!$ticketEvent->isNull()) {
+            $ticketEventArray = $ticketEvent->toArray();
+            $userArray = $user->toArray();
 
-        if(!Controller::isStaffLogged() &&  $user->id !== $ticketEventArray["author"]["id"] && $user->id !== $ticket->authorId) {
-            throw new RequestException(ERRORS::NO_PERMISSION);
-        }
+            if($user->id !== $ticketEventArray["author"]["id"] && $user->id !== $ticket->authorId) {
+                throw new RequestException(ERRORS::NO_PERMISSION);
+            }
 
-        if (!$ticketEvent->isNull()) {
-            if($user->id !== $ticketEventArray["author"]["id"]) {
+            if((!!$userArray["isStaff"] === !!$ticketEventArray["author"]["staff"]) && ($user->id !== $ticketEventArray["author"]["id"])) {
+                throw new RequestException(ERRORS::NO_PERMISSION);
+            }
+
+            if (!!$userArray["isStaff"] !== !!$ticketEventArray["author"]["staff"]) {
```

Comment:
If you write this one first, you don't need the `!!$userArray["isStaff"] === !!$ticketEventArray["author"]["staff"]` in the if from line 71.